contracts/VestingEscrowFactory.vy

```diff
@@ -1,79 +1,266 @@
-# @version 0.2.16
+# @version 0.3.7
+
 """
 @title Vesting Escrow Factory
-@author Curve Finance, Yearn Finance
+@author Curve Finance, Yearn Finance, Lido Finance
 @license MIT
-@notice Stores and distributes ERC20 tokens by deploying `VestingEscrowSimple` contracts
+@notice Stores and distributes ERC20 tokens by deploying `VestingEscrow` contracts
 """
 
 from vyper.interfaces import ERC20
 
 
-interface VestingEscrowSimple:
+interface IVestingEscrow:
     def initialize(
-        admin: address,
         token: address,
-        recipient: address,
         amount: uint256,
+        recipient: address,
         start_time: uint256,
         end_time: uint256,
         cliff_length: uint256,
+        is_fully_revokable: bool,
+        voting_adapter_addr: address,
     ) -> bool: nonpayable
 
 
 event VestingEscrowCreated:
-    funder: indexed(address)
-    token: indexed(address)
-    recipient: indexed(address)
+    creator: indexed(address)
     escrow: address
+
+
+event ERC20Recovered:
+    token: address
     amount: uint256
-    vesting_start: uint256
-    vesting_duration: uint256
-    cliff_length: uint256
+
+
+event ETHRecovered:
+    amount: uint256
+
+
+event VotingAdapterUpgraded:
+    voting_adapter: address
+
+
+event OwnerChanged:
+    owner: address
+
+
+event ManagerChanged:
+    manager: address
 
 
 target: public(address)
+token: public(address)
+voting_adapter: public(address)
+owner: public(address)
+manager: public(address)
+
 
 @external
-def __init__(target: address):
+def __init__(
+    target: address,
+    token: address,
+    owner: address,
+    manager: address,
+    voting_adapter: address,
+):
     """
     @notice Contract constructor
-    @dev Prior to deployment you must deploy one copy of `VestingEscrowSimple` which
-         is used as a library for vesting contracts deployed by this factory
-    @param target `VestingEscrowSimple` contract address
+    @dev Prior to deployment you must deploy one copy of `VestingEscrowSimple` and `VestingEscrowFullyRevokable` which
```

Comment:
This will increase code complexity a bit (for immutable in Vyper we need a standalone view method) and flexibility (current impl allows using one VestingEscrow proto for multiple factories with different tokens and owners)